### PR TITLE
[TACHYON-702] Use mUserId as HashCode

### PR DESCRIPTION
**Pull request by rogerdai16**
Why not override equals and hashCode to use mUserId to identify UserInfo object.
As mUserId is used to identify an UserInfo object.

### CODE REVIEW

**Comment by apc999** [comment]
Jenkins, please test this.

File: servers/src/main/java/tachyon/UserInfo.java [hunk 1]
```diff
@@ -54,4 +54,17 @@ public synchronized String toString() {
     sb.append(")");
     return sb.toString();
   }
+  
+  @Override
+    public boolean equals(Object o) {
```
Comment (apc999) [inline servers/src/main/java/tachyon/UserInfo.java]: wrong indentation for `equals`

**Comment by apc999** [comment]
Jenkins, test this please.

**Comment by calvinjia** [comment]
@rogerdai16 
Thanks for the patch, please open a JIRA ticket here: https://tachyon.atlassian.net and also tag your PR with the ticket number like `[TACHYON-###] Update UserInfo`.
Also, it seems like the patch does not pass the check style, could you please modify the style?

**Comment by rogerdai16** [comment]
@calvinjia 
Yes, I fixed the indentation and also changed the patch's name. 
Please check it out.

File: servers/src/main/java/tachyon/UserInfo.java [hunk 2]
```diff
@@ -54,4 +54,17 @@ public synchronized String toString() {
     sb.append(")");
     return sb.toString();
   }
+  
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
```
Comment (apc999) [inline servers/src/main/java/tachyon/UserInfo.java]: Quote from Google java style (which is adopted by Tachyon with slight modification)
>Braces are used with if, else, for, do and while statements, even when the body is empty or contains only a single statement.

**Comment by rogerdai16** [comment]
@apc999 
Now braces are fixed.

**Comment by calvinjia** [comment]
@rogerdai16 
Is this change for future benefit, I don't think we are using `UserInfo` as a key anywhere?

**Comment by rogerdai16** [comment]
@calvinjia 
Yes, there is Map<Long, UserInfo> mUsers in users.java. It actually uses mUserId as key.

**Comment by gpang** [comment]
ok to test

**Comment by apc999** [comment]
+1 LGTM

**Comment by calvinjia** [comment]
+1 LGTM

**Pull request merged**